numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1052,9 +1050,11 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
             goto fail;
         }
 
+        PyArray_FailUnlessWriteable(out, "output array");
```

Comment:
Can you please add a test for this, because this doesn't work, you need an error return check and deal with that (this is not C++ with exceptions).

Reply:
Ah yes sorry, fixed this, adding a test now.

Reply:
Test is pushed.